griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -80,11 +80,58 @@ def load_entries(
             )
 
     def query(
-        self,
-        query: str,
-        count: Optional[int] = None,
-        namespace: Optional[str] = None,
-        include_vectors: bool = False,
-        **kwargs
+            self,
+            query: str,
+            count: Optional[int] = None,
+            namespace: Optional[str] = None,
+            include_vectors: bool = False,
+            offset: Optional[int] = 0,
+            **kwargs
     ) -> list[BaseVectorStoreDriver.QueryResult]:
-        raise NotImplementedError("Vector search not supported in MongoDB")
+        collection = self.get_collection()
+
+        # Using the embedding driver to convert the query string into a vector
+        vector = self.embedding_driver.embed_string(query)
+        print("Vector for query:", vector)
+
+        knn_k = count if count else 10
+        pipeline = [
+            {
+                "$search": {
+                    "index": "knn",  # Use the name of the k-NN index you created
```

Comment:
Why is it `knn`? According to the [docs](https://www.mongodb.com/docs/atlas/atlas-search/knn-beta/#syntax) the default index name is `default`. I think we should add another optional method parameter `index: Optional[str] = None` to the method and only set it in this dictionary if it's not `None`.